main/src/cgeo/geocaching/filters/core/StringFilter.java

```diff
@@ -131,6 +134,9 @@ public void addToSql(final SqlBuilder sqlBuilder, final String columnExpression)
                 case IS_PRESENT:
                     sqlBuilder.addWhere(columnExpression + " IS NOT NULL AND " + columnExpression + " <> ''");
                     break;
+                case DOES_NOT_CONTAIN:
+                    sqlBuilder.addWhere("NOT " + getRawLikeSqlExpression(columnExpression));
```

Comment:
Might be more efficient to construct a "not like" SQL operator instead of inverting a "like" operator. But maybe it doesn't matter.

Reply:
Could you show me how a "not like" SQL operator does look like?

I do very well understand the basic SQL concepts and example queries but unfortunately have never really worked with SQL actively...

Reply:
A like operator looks like this:

`Select * from gc_cache where name like '%beer%'`

A not like operator looks like this:

`Select * from gc_cache where name not like '%beer%'`

Your change will make the SQL like this:

`Select * from gc_cache where not name like '%beer%'`

It is probably a minor, neglectable thing

Reply:
> Select * from gc_cache where not name like '%bier%'

Wouldn't this select return all colums but not those where name is '%bier%'

Reply:
> > Select * from gc_cache where not name like '%bier%'
> 
> Wouldn't this select all colums but not those where name is '%bier%'

It would select all **rows** except the ones containing 'bier' in the name column - which would be the intended behavior of "does not contain".

Note that the actual SQL statement used by c:geo is far more complex (e.g. columns are selected explicitly, not using `*`. Also the function `lower()` is used to allow for case-insensitive select). The statement examples should just explain the `NOT LIKE` operator in contrast to a negated `LIKE` operator.

Reply:
Ok, here we go. Thanks for the suggestion, looks IMHO also codewise much cleaner...

@eddiemuc Would you mind another quick cross-check?

Reply:
LGTM